core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -587,7 +587,7 @@ public void executeTask(KeyValueFilter<? super WrappedBytes, ? super WrappedByte
 
    @Override
    public Iterator<InternalCacheEntry<WrappedBytes, WrappedBytes>> iterator() {
-      long time = timeService.time();
+      long time = timeService.wallClockTime();
```

Comment:
Not this test :) This test is all because of the system clock rolling over or not. I am able to reproduce it now, because my system clock has rolled over.